mbuild/lib/recipes/polymer.py

```diff
@@ -1,75 +1,296 @@
 import itertools as it
 
+import numpy as np
+
+from mbuild.port import Port
 from mbuild.compound import Compound
 from mbuild.coordinate_transform import force_overlap
+from mbuild.lib.atoms import H
 from mbuild.utils.validation import assert_port_exists
 from mbuild import clone
 
 
-__all__ = ['Polymer']
+__all__ = ["Polymer"]
 
 
 class Polymer(Compound):
     """Connect one or more components in a specified sequence.
 
-    Parameters
+    Attributes
     ----------
-    monomers : mb.Compound or list of mb.Compound
-        The compound(s) to replicate.
-    n : int
-        The number of times to replicate the sequence.
-    sequence : str, optional, default='A'
-        A string of characters where each unique character represents one
-        repetition of a monomer. Characters in `sequence` are assigned to
-        monomers in the order assigned by the built-in `sorted()`.
-    port_labels : 2-tuple of strs, optional, default=('up', 'down')
-        The names of the two ports to use to connect copies of proto.
+    monomers : List of mbuild.Compounds
+        The compound(s) to replicate. Add to this list using the add_monomers
+        method.
+    end_groups : List of mbuild.Compounds
+        The compound to cap the end of the polymer. Add to this list using the
+        add_end_groups method.
+
+    Methods
+    -------
+    add_monomer(monomer, indices, separation, port_labels, orientation, replace)
+        Use to add a monomer compound to Polymer.monomers
+
+    add_end_groups(compound, index, separation, orientation, replace)
+        Use to add an end group compound to Polymer.end_groups
 
+    build(n, sequence)
+        Use to create a single polymer compound. This method uses the compounds
+        created by calling the add_monomer and add_end_group methods.
     """
-    def __init__(self, monomers, n, sequence='A', port_labels=('up', 'down')):
-        if n < 1:
-            raise ValueError('n must be 1 or more')
+
+    def __init__(self, monomers=None, end_groups=None):
         super(Polymer, self).__init__()
-        if isinstance(monomers, Compound):
-            monomers = (monomers,)
-        for monomer in monomers:
-            for label in port_labels:
+        self._monomers = monomers or []
+        self._end_groups = end_groups or [None, None]
+        if len(self._end_groups) != 2:
+            raise ValueError(
```

Comment:
I think once you add some unit tests to ensure that these errors are being properly raised when encountered, this should be good to go!

Reply:
Oops, I added some tests, but not ones that check for these errors.  I'll be sure to add these as well.